### PR TITLE
Domain admin tests in roster suite

### PR DESCRIPTION
**Pull request by jacekwegr**
Added tests for domain admin in graphql_roster_SUITE. The tests are similar to those for a global admin, but they also check for permissions.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/132999) / small_tests / 721e9f8b8d94f16a200e38c01858df5693083f74
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3736/132999/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3736/132999/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-08-18_14.47.23/lib.mongooseim.logs/run.2022-08-18_14.47.24/suite.log.html)

File: big_tests/tests/graphql_roster_SUITE.erl [hunk 1]
```diff
@@ -499,6 +522,127 @@ user_get_nonexistent_contact_story(Config, Alice) ->
...
+
+domain_admin_set_mutual_subscription_try_connect_nonexistent_users(Config) ->
+    Alice = ?NONEXISTENT_DOMAIN_USER,
+    Bob = ?NONEXISTENT_USER,
+    Res = domain_admin_mutual_subscription(Alice, Bob, <<"CONNECT">>, Config),
+    get_unauthorized(Res).
+
+domain_admin_set_mutual_subscription_try_connect_users_no_permission(Config) ->
+    escalus:fresh_story_with_config(Config, [{alice_bis, 1}, {bob, 1}],
+                                    fun domain_admin_set_mutual_subscription_try_connect_users_no_permission_story/3).
```
Comment (JanuszJakubiec) [inline big_tests/tests/graphql_roster_SUITE.erl:626]:
```suggestion
        fun domain_admin_set_mutual_subscription_try_connect_users_no_permission_story/3).
```

Comment (JanuszJakubiec) [reply big_tests/tests/graphql_roster_SUITE.erl:626]: Please try to stick to the 100 characters per line limit.

**Review by JanuszJakubiec — CHANGES_REQUESTED** [review]
Overall looks good, however, I think that copying all tests for domain_admin tests is unnecessary.  A sufficient amount would be to test each graphql function two times for domain admin(one when it passes and one when an authentication error occurs). More details I will give later during the call.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/133078) / small_tests / 4367b5f8b8f5efaaef1615d8855ea372966642c3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3736/133078/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3736/133078/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-08-19_08.26.10/lib.mongooseim.logs/run.2022-08-19_08.26.11/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/133105) / small_tests / c368c5109f54725b598f44d0f6fa3e1daab859b2
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3736/133105/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3736/133105/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-08-19_14.45.19/lib.mongooseim.logs/run.2022-08-19_14.45.20/suite.log.html)

**Comment by mongoose-im** [comment]
[riak_mnesia_24](https://circleci.com/gh/esl/MongooseIM/133699) / riak_mnesia / b10550ed2fe5069389ad5de7bba036b6acdae930
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3736/133699/riak_mnesia.24.3.4)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/133879) / small_tests / 44684acf636fef1a13f3d18697c5eeb40efdc342
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3736/133879/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3736/133879/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-08-26_07.49.18/lib.mongooseim.logs/run.2022-08-26_07.49.18/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/133935) / small_tests / 6aca4dd512fd195bc3746d2deba7c32ce53cbb2f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3736/133935/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3736/133935/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-08-26_09.28.34/lib.mongooseim.logs/run.2022-08-26_09.28.34/suite.log.html)

File: big_tests/tests/graphql_roster_SUITE.erl [hunk 2]
```diff
@@ -513,6 +591,60 @@ domain_admin_subscribe_all_to_all_no_permission(Config, Alice, Bob, Kate) ->
...
+domain_admin_subscribe_to_all_with_wrong_user(Config) ->
+    escalus:fresh_story_with_config(Config, [{alice, 1}, {bob, 1}],
+                                    fun domain_admin_subscribe_to_all_with_wrong_user_story/3).
+
+domain_admin_subscribe_to_all_with_wrong_user_story(Config, Alice, Bob) ->
+    Kate = ?NONEXISTENT_DOMAIN_USER,
+    Res = admin_subscribe_to_all(Alice, [Bob, Kate], Config),
+    check_if_created_succ(?SUBSCRIBE_TO_ALL_PATH, Res, [true, false]),
+    ?assertNotEqual(nomatch, binary:match(get_err_msg(Res), <<"does not exist">>)),
+
```
Comment (chrzaszcz) [inline big_tests/tests/graphql_roster_SUITE.erl:612]: I would remove this empty line

**Review by chrzaszcz — APPROVED** [review]
Looks good. I like that all cases are checked, even the obvious ones (e.g. for existing and non-existent user/domain combinations). This is more than I expected, but it's almost always good to test more.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/135621) / small_tests / ecd2cf364ba2ef0764aab2031da7683bf5d38b46
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3736/135621/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3736/135621/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-08_07.15.31/lib.mongooseim.logs/run.2022-09-08_07.15.31/suite.log.html)

**Review by JanuszJakubiec — APPROVED** [review]

**Pull request merged**